src/include/sof/bit.h

```diff
@@ -13,6 +13,23 @@
 #undef BIT
 #endif
 
+#ifdef SET_BIT
+#undef SET_BIT
+#endif
+
+#ifdef SET_BITS
+#undef SET_BITS
+#endif
+
+#ifdef GET_BIT
+#undef GET_BIT
+#endif
+
+#ifdef GET_BITS
+#undef GET_BITS
+#endif
```

Comment:
Found this via https://github.com/zephyrproject-rtos/zephyr/pull/48963

This looks mildly dangerous to me.  That resulting SOF code is going to get either the Zephyr *or* the SOF versions of these macros essentially arbitrarily, depending on header inclusion order.

Better to find the SOF definitions and put them under an `#ifndef __ZEPHYR__`, adding a `#include <zephyr/sys/util.h>` (or wherever these macros are) to the same file.

Reply:
The problem is that this would require Zephyr to have all these macros defined there and if we change some name it will break SoF. To make it deterministic I can include zephyr header in this file, and keep this conditionals, this way we ensure that the inclusion order doesn't  matter.

Reply:
My preference here is to go full native (i.e. use the Zephyr macros directly), and only redefine locally when building for xtos (and this should be in a separate area).

Reply:
we don't have all these macros on Zephyr and will not have. For example Zephyr has a different implementation for `MASK`. It means that we will need a slightly bigger change in SoF. Let me look it.